### ISSUE TITLE
Add customization which appends builder to the `Customizations` collection

### ISSUE THREAD

**Issue by zvirja** [AutoFixture/AutoFixture#1056]
When you would like to build `CustomizeAttribute` instance (extension point for NUnit and xUnit), you should return the `ICustomization` instance:
```c#
public class CustomAttribute : CustomizeAttribute
{
    public override ICustomization GetCustomization(ParameterInfo parameter)
    {
        // Here I should return ICustomization
    }
}
```

I found that often all I need to do is to insert `ISpecimenBuilder` to the `fixture.Customizations` collection. However, to do that I need to create an extra class, implementing the `ICustomization` interface and adding my builder to the collection. It makes implementation polluted and it's hard to focus on the logic.

I suggest to add this type to the AutoFixture library, so we could write simpler customizations:
```c#
public class FortyTwo : CustomizeAttribute
{
    public override ICustomization GetCustomization(ParameterInfo parameter)
    {
        return new AddCustomBuilderCustomization(
            new FilteringSpecimenBuilder(
                new FixedBuilder(42),
                new EqualRequestSpecification(parameter)
            ));
    }
}
```

The only thing I'm unsure is the customization name. Usually we follow the convention to suffix customizations with `Customization` word, but here it might add noise. I see following candidates:
- `AddCustomBuilderCustomization`
- `AddBuilderToCustomizations`
- `AddBuilderToCustomizationsCustomization`
- `CustomizationAppender`
- `CustomizationAppenderCustomization`

None of the name looks great 😞 Also I don't like the "append" part, as later we might want to specify the position (first vs last), so append will be confusing.

@moodmosaic Do you see a nice name for this customization type?

**Comment by moodmosaic** [comment]
I'd do something like this:

```c#
public class FortyTwo : CustomizeAttribute
{
    public override ICustomization GetCustomization(ParameterInfo parameter)
    {
        return
            new FilteringSpecimenBuilder(
                new FixedBuilder(42),
                new EqualRequestSpecification(parameter))
            .AsCustomization();
    }
}
```

**Comment by zvirja** [comment]
@moodmosaic Wow, such a nice solution! Thanks :relaxed:

**Comment by Kralizek** [comment]
I don't understand the concrete use case and where I would use this. ELI5? :)

**Comment by zvirja** [comment]
@Kralizek I'd suggest to re-read the issue description - I've described the  _concrete_ use case there :relaxed: It's not an everyday's scenario, but it's still quite often.

**Comment by Kralizek** [comment]
Yeah, read it twice or thrice and still couldn't figure out which kind of scenario you find yourself in.

Don't take it as a review of the issue or of the implementation. I'm really just trying to learn more about the customization process.

**Comment by zvirja** [comment]
@Kralizek Consider you have the following xUnit test:
```c#
public class Issue1056
{
    [Theory, AutoData]
    public void SomeTest(SystemUnderTest sut, [Negative] int negValue)
    {
        var result = sut.GetAbs(negValue);
        Assert.Equal(-negValue, result);
    }
    
    public class SystemUnderTest
    {
        public int GetAbs(int value) => Math.Abs(value);
    }
    
    public class Negative : CustomizeAttribute
    {
        public override ICustomization GetCustomization(ParameterInfo parameter)
        {
            return new FilteringSpecimenBuilder(
                    new FixedBuilder(-10),
                    new EqualRequestSpecification(parameter)
                ).AsCustomization();
        }
    }
}
```

You want to customize the test input value, so that fixture provides you with a negative number (by default the positive numbers are returned). To do that you can write your own customization attribute. As a part of the customization attribute implementation you should return the `ICustomization` instance. Here is where the extension method is very handy, as otherwise you need to create an extra class.

Sure, the real-world implementation should be more clever, but I just wanted to demonstrate you the idea 😅

**Comment by Kralizek** [comment]
Oh nice! Thank you for your explanation! I already see some "arrange" code that can be moved away from the unit test body!

The more I learn about AutoFixture, the more I love it.
